data/blog/search.md

```diff
@@ -0,0 +1,162 @@
+be difficult to learn best practices, find the right APIs, or discover the tools 
+that will help you build the app of your dreams. As app developers, we often find 
+ourselves turning to sites like  Google and Stack Overflow for answers. This 
+works, but we can do better.
+
+Today we're rolling out a new search feature on the Electron website that will
+help developers find the resources they need to build apps faster and more 
+easily.
+
+To try it out, visit any page on [electronjs.org](https://electronjs.org)
+and you'll find a search input at the top of the page.
+
+## The Search Engine
+
+When we first set about adding search to the website, we rolled our own
+search engine using GraphQL as a backend. GraphQL was fun to work with and
+the search engine was performant, but we quickly realized that building a search
+engine is not a trivial task. Things like multi-word search and typo detection
+require a lot of work to get right. Rather than reinventing the wheel, 
+we decided to use an existing search solution: [Algolia].
+
+Algolia is a hosted search service that has quickly become the
+search engine of choice among popular open source projects like 
+React, Vue, Bootstrap, Yarn, and [many others](https://community.algolia.com/docsearch/).
+
+Here are some of the features that made Algolia a good fit for the Electron project:
+
+- [InstantSearch.js](https://community.algolia.com/instantsearch.js) provides results as you type, usually in about 1ms.
+- [Typo tolerance](https://www.algolia.com/doc/guides/textual-relevance/typo-tolerance/) means you'll still get results even when you type [`widnow`].
+- [Advanced query syntax](https://www.algolia.com/doc/api-reference/api-parameters/advancedSyntax/) enables `"exact quoted matches"` and `-exclusion`.
+- [API clients](https://www.algolia.com/doc/api-client/javascript/getting-started/) are open source and with well-documented.
+- [Analytics](https://www.algolia.com/doc/guides/analytics/analytics-overview/) tell us what people are searching for most, as well as what they're searching for but not finding. This will give us valuable insight into how Electron's documentation can be improved.
+- Algolia is [free for open source projects](https://www.algolia.com/for-open-source).
+
+## API Docs
+
+Sometimes you know *what* you want to accomplish, but you don't know exactly 
+*how* to do it. Electron has over 750 API methods, events, and properties. 
+No human can easily remember all of them, but computers are good at this stuff. 
+Using Electron's [JSON API docs](https://electronjs.org/blog/api-docs-json-schema),
+we indexed all of this data in Algolia, and now you can easily find
+the exact API you're looking for.
+
+Trying to resize a window? Search for [`resize`] and jump straight to the method you need.
+
+## Tutorials
+
+Electron has an ever-growing collection of tutorials to complement its API
+documentation. Now you can more easily find all tutorials on a given topic,
```

Comment:
I'd remove "all" since presumably there are offsite tutorials that we and Algolia aren't aware of?